ui/webui/src/components/storage/CustomMountPoint.jsx

```diff
@@ -0,0 +1,276 @@
+                <HelperText>
+                    <HelperTextItem variant="error" hasIcon>
+                        {_("Duplicate mount point.")}
+                    </HelperTextItem>
+                </HelperText>}
+        </Flex>
+    );
+};
+
+const MountpointCheckbox = ({ reformat, isRootMountPoint, handleCheckReFormat, partition, isDisabled }) => {
+    return (
+        <Flex>
+            <Checkbox
+              label={_("Format")}
+              isChecked={reformat}
+              isDisabled={isRootMountPoint || isDisabled}
+              onChange={(checked, _) => handleCheckReFormat(checked, partition)}
+              id={partition}
+            />
+            {isRootMountPoint &&
+                <Popover
+                  bodyContent={_("The root partition is always re-formatted by the installer.")}
+                  showClose={false}>
+                    <HelpIcon />
+                </Popover>}
+        </Flex>
+    );
+};
+
+export const CustomMountPoint = ({ deviceData, idPrefix, setIsFormValid, onAddErrorNotification, toggleContextHelp, stepNotification }) => {
+    const [requests, setRequests] = useState(null);
+    const [partitioning, setPartitioning] = useState(null);
+
+    const validateMountPoints = requests => {
+        if (requests !== null) {
+            const mountPoints = requests.map(r => r["mount-point"]);
+            setIsFormValid(new Set(mountPoints).size === mountPoints.length);
+        }
+    };
+
+    validateMountPoints(requests);
+
+    useEffect(() => {
+        if (requests === null) {
+            findPartitioning({ method: "MANUAL" }).then(([partitioning]) => {
+                setPartitioning(partitioning);
+                return gatherRequests({ partitioning });
+            })
+                    .then(([res]) => {
+                        setRequests(res
+                                .map(row => {
+                                    return {
+                                        "device-spec": row["device-spec"].v,
+                                        "format-type": row["format-type"].v,
+                                        "mount-point": row["mount-point"].v,
+                                        reformat: row.reformat.v
+                                    };
+                                }));
+                    })
+                    .catch(console.error);
+        }
+    });
+
+    const requestsToDbus = requests => {
+        return requests.map(row => {
+            return {
+                "device-spec": cockpit.variant("s", row["device-spec"]),
+                "format-type": cockpit.variant("s", row["format-type"]),
+                "mount-point": cockpit.variant("s", row["mount-point"]),
+                reformat: cockpit.variant("b", row.reformat),
+            };
+        });
+    };
+
+    const handleOnSelect = (selection, device) => {
+        const newRequests = requests.map(row => {
```

Comment:
i would like to emit the requests object in the state and read it from the global state instead.